neuro_san/internals/messages/intra_agent_message_utils.py

```diff
@@ -0,0 +1,98 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+from typing import Any
+from typing import List
+from typing import Union
+
+import json
+
+from langchain_core.messages.base import BaseMessage
+
+from neuro_san.internals.messages.chat_message_type import ChatMessageType
+
+
+class IntraAgentMessageUtils:
+    """
+    Utility class to parse messages with role information.
+    This ancient style of message is used between agents and tool calls
+    and is agnostic to OpenAI vs langchain.
+    """
+
+    @staticmethod
+    def generate_response(the_messages: List[Any]) -> str:
```

Comment:
This comes from message_utils.py and is made a static method on a class.

The contents of this method have been refactored a bit for clarity and for fewer calls to the same methods.